docs/2.guide/2.directory-structure/1.server.md

```diff
@@ -283,12 +283,12 @@ export default fromNodeMiddleware((req, res, next) => {
 ```
 
 ::alert{type=warning}
-Never combine `next()` callback with a legacy middleware that is `async` or returns a `Promise`!
+Never combine the `next()` callback with a legacy middleware that is `async` or returning a `Promise`!
```

Comment:
_that returns a `Promise` or is `async`_

Knowing the above example seems to sound correct, shall we keep the original, inversed version with _returns_? Otherwise, this would mean **that it is `async`** and **it is returning a `Promise`**. Not sure whether we need Present Continuous tense here.

It can be logically split in the following way:

> (...) a legacy middleware that:
> 
> - is async
> - returns a `Promise`
```suggestion
Never combine the `next()` callback with a legacy middleware that is async or returns a `Promise`!
```